### PR TITLE
 input loop handles steps, replaced spawn() loopage

### PR DESCRIPTION
**Pull request by mybluecorners**
i need people to test this and compare with the old spawn loop - to make sure they behave EXACTLY THE SAME before we switch

### CODE REVIEW

**Comment by ZeWaka** [comment]
Doo eeet (after merge)

File: code/input.dm [hunk 1]
```diff
@@ -329,12 +330,11 @@ var/list/dirty_keystates = list()
 			var/next = world.time + actual_delay
 			var/lmt = max(last_move_ticklag - world.tick_lag, 0)
```
Comment (pali6) [inline code/input.dm:331]: Unused variable. Should be probably either removed or used. :shelterfrog:

**Comment by pali6** [comment]
How about we test merge this for a single round and ask players what they think. The test version could even be modified so various knobs can be turned live to see if they improve things or not.

**Comment by ZeWaka** [comment]
> How about we test merge this for a single round and ask players what they think.

This, but don't tell them.

**Pull request merged**